apis/python/src/tiledb/vector_search/object_readers/soma_reader.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any, Mapping, Optional, List, Dict, OrderedDict, Tuple
+# from tiledb.vector_search.object_readers import ObjectPartition, ObjectReader
+from tiledb import Attr
+
+# class SomaPartition(ObjectPartition):
+class SomaPartition():
+    def __init__(
+        self,
+        partition_id: str,
+        coord_start: int,
+        coord_end: int,
+        **kwargs,
+    ):
+        self.partition_id = partition_id
+        self.coord_start = coord_start
+        self.coord_end = coord_end
+        self.size = coord_end - coord_start
+
+    def init_kwargs(self) -> Dict:
+        return {
+            "partition_id": self.partition_id,
+            "coord_start": self.coord_start,
+            "coord_end": self.coord_end,
+        }
+
+    def num_vectors(self) -> int:
+        return self.size
+
+    def index_slice(self) -> Tuple[int,int]:
+        return (self.coord_start, self.coord_end)
+
+
+# class SomaReader(ObjectReader):
+class SomaReader():
```

Comment:
(Optional) Should we be more precise in the naming here so that we don't make people think this is a generic soma reader? I.e. something like `SomaRNAXAxisReader`.

Reply:
Changed